frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInBranchBelowAction.java

```diff
@@ -0,0 +1,219 @@
+      log().debug("Skipping action: Branch layout entry already exists below given parent");
+      return;
+    }
+
+    new Task.Backgroundable(project, GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.task-title")) {
+
+      @Override
+      public void run(ProgressIndicator indicator) {
+        Path macheteFilePath = getMacheteFilePath(selectedVcsRepository);
+
+        var entryToSlideIn = branchLayout.findEntryByName(slideInDialogBranchName)
+            .getOrElse(new SlideInEntry(slideInDialogBranchName, /* customAnnotation */ null, /* children */ List.empty()));
+
+        var newBranchLayout = Try
+            .of(() -> branchLayout.slideIn(parentName, entryToSlideIn))
+            .onFailure(
+                t -> notifier.notifyError(
+                    /* title */ GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.notification.fail",
+                        slideInDialogBranchName),
+                    getMessageOrEmpty(t)))
+            .toOption();
+
+        newBranchLayout.map(nbl -> Try.run(() -> branchLayoutWriter.write(macheteFilePath, nbl, /* backupOldLayout */ true))
+            .onFailure(t -> notifier.notifyError(
+                /* title */ GitMacheteBundle
+                    .message("action.GitMachete.BaseSlideInBranchBelowAction.notification.fail.branch-layout-write"),
+                getMessageOrEmpty(t))));
+      }
+    }.queue();
+  }
+
+  private static String getMessageOrEmpty(Throwable t) {
+    return t.getMessage() != null ? t.getMessage() : "";
+  }
+
+  @Nullable
+  String createOrCheckoutNewBranch(Project project, GitRepository gitRepository, String startPoint, String initialName) {
+    var repositories = List.of(gitRepository).asJava();
+    var gitNewBranchDialog = new GitNewBranchDialog(project,
+        repositories,
+        /* title */ GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.dialog.create-new-branch.title",
+            startPoint),
+        initialName,
+        /* showCheckOutOption */ true,
+        /* showResetOption */ true,
+        /* localConflictsAllowed */ true);
+
+    var options = gitNewBranchDialog.showAndGetOptions();
+
+    if (options == null) {
+      log().debug(
+          "Name of branch to slide in is null: most likely the action has been canceled from create-new-branch dialog");
+      return null;
+    }
+
+    var branchName = options.getName();
+    if (!initialName.equals(branchName)) {
+      return branchName;
+    }
+
+    if (options.shouldCheckout()) {
+      checkoutOrReset(project, repositories, startPoint, options);
+    } else {
+      createNewBranch(project, repositories, startPoint, options);
+    }
+
+    return options.getName();
+  }
+
+  @AllArgsConstructor
+  @SuppressWarnings("interning:not.interned") // to allow for `==` comparison in Lombok-generated `withChildren` method
+  private static final class SlideInEntry implements IBranchLayoutEntry {
```

Comment:
Do we still need this one? Can we use `BranchLayoutEntry` instead?

Reply:
Oh ok, `BranchLayoutEntry` isn't available in the API

Reply:
Hmm not sure if I brought this up before... but you could consider exposing `BranchLayoutEntry` in the API as well (and then here an extra class won't be needed)

Reply:
dropped revert XD it is in the API again